### DISCUSSION TITLE
Export/Download table data in V8

### DISCUSSION THREAD

**Discussion by tejasvis** [TanStack/table#5118] [Q&A]
Hello everyone,

How to export/download table data from the table instance. I can pull up the columns from getVisibleColumns(), but it's had for row data to associate them with columns. I know it was simpler and direct in V7 but can't figure it out in V8. _valuesCache isn't helpful with pagination in place. I use accessorFn across every reference , and it is so hard to match them with the keys returning in row.original. Is there any other way?

**Comment by KevinVandy** [comment]
Simply use any of the row models. I have a couple examples on my site: https://www.mantine-react-table.com/docs/examples/export-csv

**Reply by tejasvis** [reply]
Yes that's what I have been using, but I was looking for more a better direct way to get row data like an object of 
**< Column Header >:<Cell value>** instead **< Column Accessor key/Fn>:<Cell value>** like in your examples. I remember V7 has a direct way to just pull row data like that. I was wondering if v8 has a way or do I have to write an alternate approach to replace accessorKeys.

row. Original looks like this :
<img width="845" alt="image" src="https://github.com/TanStack/table/assets/23281194/01fdfc7a-4559-4a53-9aa9-465f4025d949">


but I'm looking for like this(can't use valuesCache as it wouldn't render everything on a paginated data):

<img width="796" alt="image" src="https://github.com/TanStack/table/assets/23281194/0b7b7b30-3062-4bb4-b920-199a2e5eced4">

**Reply by KevinVandy** [reply]
Just map over each cell and use cell.getValue() if you don't want to use row.original for some reason.

**Reply by aertyuiop12345** [reply]
I've seen it on your site, but isn't it available in vuejs with tanstack?

**Reply by mbauchet** [reply]
Hey @KevinVandy, how can I export the value of columns with "cell" render?
